### PR TITLE
Changed facility feature placement and added a generic tooltip component

### LINKED ISSUE #3176: Facility Faetures
The Facilities now have icons for the features in the facility in the facility list
Screenshot 2022-07-21 at 5 57 08 PM

However, its difficult to know what the icon denotes.

When the user hovers over the icon, display the feature in text. 
In the above image, it should be "Maternity Care" in a floating box. 
Alternatively, you can also make the icon label expand to have the text when the user hovers over the icon.

### PR DESCRIPTION
**Pull request by skks1212**
closes #3176 
image

### CODE REVIEW

**Review by Ashesh3 — APPROVED** [review]

File: src/Components/Common/utils/Tooltip.tsx [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import clsx from "clsx";
+import { ReactNode, useState } from "react"
+
+export default function ToolTip(props : {
+    children : ReactNode
+    text : ReactNode
+    position : "TOP" | "BOTTOM" | "LEFT" | "RIGHT"
```
Comment (developedBySJ) [inline src/Components/Common/utils/Tooltip.tsx]: You can make it as an optional prop and use `ToolTip.defaultProps = { position : "TOP" } ` for default value

File: src/Components/Common/utils/Tooltip.tsx [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+}){
+
+    const [status, show] = useState(false);
+
+    const style = clsx({
+        "absolute bg-black/50 backdrop-blur rounded text-white transition px-2 py-1 z-50 w-[150px] text-center block" : true,
```
Comment (developedBySJ) [inline src/Components/Common/utils/Tooltip.tsx]: You can remove `true` as it is not needed
```suggestion
        "absolute bg-black/50 backdrop-blur rounded text-white transition px-2 py-1 z-50 w-[150px] text-center block",
```

Comment (shivankacker) [reply src/Components/Common/utils/Tooltip.tsx]: Hey, this will be required as this is an object

**Review by cp-Coder — APPROVED** [review]

**Comment by nihal467** [comment]
@skks1212 
image

the UI is not looking good, is there a alternative approach to it ?

**Comment by shivankacker** [comment]
> @skks1212 image
> 
> the UI is not looking good, is there a alternative approach to it ?

@nihal467 I don't think there is any other approach to this other than tooltips... do you have any suggestions?

**Comment by nihal467** [comment]
test approved

File: src/Components/Common/utils/Tooltip.tsx [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+        "bottom-[calc(100%+5px)] left-[calc(50%-75px)]" : position === "TOP",
+        "top-[calc(100%+5px)] left-[calc(50%-75px)]" : position === "BOTTOM",
+        "right-[calc(100%+5px)] top-[calc(50%-75px)]" : position === "LEFT",
+        "left-[calc(100%+5px)] top-[calc(50%-75px)]" : position === "RIGHT",
+        "visible opacity-100 -translate-y-1" : status === true,
+        "invisible opacity-0 translate-y-0" : status === false
+    })
```
Comment (khavinshankar) [inline src/Components/Common/utils/Tooltip.tsx]:
```suggestion
    const style = clsx({
        "absolute bg-black/50 backdrop-blur rounded text-white transition px-2 py-1 z-50 w-[150px] text-center block",
        position === "TOP" && "bottom-[calc(100%+5px)] left-[calc(50%-75px)]",
        position === "BOTTOM" && "top-[calc(100%+5px)] left-[calc(50%-75px)]",
        position === "LEFT" && "right-[calc(100%+5px)] top-[calc(50%-75px)]",
        position === "RIGHT" && "left-[calc(100%+5px)] top-[calc(50%-75px)]",
        status === true 
        ? "visible opacity-100 -translate-y-1"
        : "invisible opacity-0 translate-y-0"
    })
```

**Review by khavinshankar — APPROVED** [review]

File: src/Components/Common/utils/Tooltip.tsx [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    const style = clsx({
+        "absolute bg-black/50 backdrop-blur rounded text-white transition px-2 py-1 z-50 w-[150px] text-center block",
+        position === "TOP" && "bottom-[calc(100%+5px)] left-[calc(50%-75px)]",
+        position === "BOTTOM" && "top-[calc(100%+5px)] left-[calc(50%-75px)]",
+        position === "LEFT" && "right-[calc(100%+5px)] top-[calc(50%-75px)]",
+        position === "RIGHT" && "left-[calc(100%+5px)] top-[calc(50%-75px)]",
+        status === true 
+        ? "visible opacity-100 -translate-y-1"
+        : "invisible opacity-0 translate-y-0"
+    })
```
Comment (khavinshankar) [inline src/Components/Common/utils/Tooltip.tsx]:
```suggestion
    const style = clsx(
        "absolute bg-black/50 backdrop-blur rounded text-white transition px-2 py-1 z-50 w-[150px] text-center block",
        position === "TOP" && "bottom-[calc(100%+5px)] left-[calc(50%-75px)]",
        position === "BOTTOM" && "top-[calc(100%+5px)] left-[calc(50%-75px)]",
        position === "LEFT" && "right-[calc(100%+5px)] top-[calc(50%-75px)]",
        position === "RIGHT" && "left-[calc(100%+5px)] top-[calc(50%-75px)]",
        status === true 
        ? "visible opacity-100 -translate-y-1"
        : "invisible opacity-0 translate-y-0"
    )
```

**Review by gigincg — APPROVED** [review]

**Pull request merged**